### PR TITLE
login/oauth/userinfo endpoint [draft]

### PR DESCRIPTION
**Pull request by 6543**
draft for #8534

### CODE REVIEW

File: routers/user/oauth.go [hunk 1]
```diff
@@ -370,6 +371,42 @@ func AccessTokenOAuth(ctx *context.Context, form auth.AccessTokenForm) {
...
+	Email             string `json:"email"`
+	Language          string `json:"language"`
+	Locale            string `json:"locale"`
+}
+
+// GetUserinfoOauth return json userinfo as in spec:
+func GetUserinfoOauth(ctx *context.Context) {
+	user := convert.ToUser(ctx.User, ctx.IsSigned, ctx.User != nil)
+	app_url := setting.AppURL
+	if setting.AppSubURL != "" {
```
Comment (6543) [inline routers/user/oauth.go]: If there exist a better way to get the **Profile-URL** just tell me ...

**Comment by lafriks** [comment]
It won't work quite like that. userinfo endpoint is called s2s so there won't be signed in user, so it has to use JWT access token. I actually also have already started to work on this :]

**Comment by 6543** [comment]
@lafriks ok ... what is your PR Nr.?

**Comment by 6543** [comment]
@lafriks ok ... what is your PR Nr.?

PS: feel free to use this code / close it ... (it's only a draft)

File: routers/user/oauth.go (hunk 1 above)
Comment (adelowo) [reply routers/user/oauth.go]: `user.HomeLink()` or `user. HTMLURL()`

Comment (6543) [reply routers/user/oauth.go]: @adelowo nice to know! thx ... but it looks like @lafriks is working on this already ... :=

**Comment by 6543** [comment]
@lafriks whats the current state of yours?

**Comment by 6543** [comment]
@lafriks " I actually also have already started to work on this :]" ... can I close mine?

**Comment by lafriks** [comment]
I will close when I submit mine :)

**Comment by sikmir** [comment]
@lafriks is there any news on your PR?

File: routers/user/oauth.go [hunk 2]
```diff
@@ -370,6 +371,42 @@ func AccessTokenOAuth(ctx *context.Context, form auth.AccessTokenForm) {
...
+		app_url = app_url + setting.AppSubURL + "/"
+	}
+	user_info := &OauthUserInfo{
+		Sub:               "",
+		Name:              user.FullName,
+		PreferredUsername: user.UserName,
+		Profile:           app_url + user.UserName,
```
Comment (6543) [inline routers/user/oauth.go]:
```suggestion
		Profile:           user.HTMLURL()
```

File: routers/user/oauth.go [hunk 3]
```diff
@@ -370,6 +371,42 @@ func AccessTokenOAuth(ctx *context.Context, form auth.AccessTokenForm) {
...
+	Picture           string `json:"picture"`
+	Email             string `json:"email"`
+	Language          string `json:"language"`
+	Locale            string `json:"locale"`
+}
+
+// GetUserinfoOauth return json userinfo as in spec:
+func GetUserinfoOauth(ctx *context.Context) {
+	user := convert.ToUser(ctx.User, ctx.IsSigned, ctx.User != nil)
+	app_url := setting.AppURL
```
Comment (6543) [inline routers/user/oauth.go]:
```suggestion
```

File: routers/user/oauth.go (hunk 1 above)
Comment (6543) [inline routers/user/oauth.go]:
```suggestion
```

File: routers/user/oauth.go [hunk 4]
```diff
@@ -370,6 +371,42 @@ func AccessTokenOAuth(ctx *context.Context, form auth.AccessTokenForm) {
...
+	if setting.AppSubURL != "" {
+		app_url = app_url + setting.AppSubURL + "/"
```
Comment (6543) [inline routers/user/oauth.go]:
```suggestion
```

File: routers/user/oauth.go [hunk 5]
```diff
@@ -370,6 +371,42 @@ func AccessTokenOAuth(ctx *context.Context, form auth.AccessTokenForm) {
...
+	}
```
Comment (6543) [inline routers/user/oauth.go]:
```suggestion
```

**Pull request closed without merging**